packages/core/src/animation/Animator.ts

```diff
@@ -783,21 +882,24 @@ export class Animator extends Component {
     layerIndex: number,
     lastClipTime: number,
     curClipTime: number
-  ) {
+  ): AnimatorStateTransition {
+    const { state } = playState;
     let transitionIndex = playState.currentTransitionIndex;
-    const duration = playState.state._getDuration();
+    const duration = state._getDuration();
     for (let n = transitions.length; transitionIndex < n; transitionIndex++) {
       const transition = transitions[transitionIndex];
       const exitTime = transition.exitTime * duration;
       if (exitTime > curClipTime) {
         break;
       }
 
-      if (exitTime >= lastClipTime) {
-        this._crossFadeByTransition(transition, layerIndex);
+      if (exitTime >= lastClipTime && this._checkConditions(state, transition)) {
```

Comment:
playState.currentTransitionIndex should update even `this._checkConditions(state, transition)` is false